plugins/test/fu-test-plugin.c

```diff
@@ -220,25 +157,38 @@ fu_test_plugin_write_firmware(FuPlugin *plugin,
 			      FwupdInstallFlags flags,
 			      GError **error)
 {
-	FuTestPlugin *self = FU_TEST_PLUGIN(plugin);
 	const gchar *test = g_getenv("FWUPD_PLUGIN_TEST");
 	gboolean requires_activation = g_strcmp0(test, "requires-activation") == 0;
 	gboolean requires_reboot = g_strcmp0(test, "requires-reboot") == 0;
-	if (g_strcmp0(test, "fail") == 0) {
+	const gchar *tmp;
+	guint64 delay_decompress_ms = 0;
+	guint64 delay_write_ms = 0;
+	guint64 delay_verify_ms = 0;
+	guint64 delay_request_ms = 0;
```

Comment:
I guess this is asking for new test suite tests for the _timing_.  When > 0 check that nothing happened before the the time and check that something happened after the time.

Reply:
I'm hesitant to do this in the tests; we have to cope with CI doing things with huge amount of skew, and also it slows the test down considerably even for 10ms delays.